src/Controls/src/Core/Platform/ImageSourceLoader.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+#if __IOS__ || MACCATALYST
+using NativeView = UIKit.UIImage;
+#elif MONOANDROID
+using NativeView = Android.Graphics.Drawables.Drawable;
+#elif WINDOWS
+using NativeView = Microsoft.UI.Xaml.Media.ImageSource;
+#elif NETSTANDARD || (NET6_0 && !IOS && !ANDROID)
+using NativeView = System.Object;
+#endif
+
+namespace Microsoft.Maui.Controls.Platform
+{
+	partial class ImageSourceLoader : IImageSourcePart
+	{
+		public IImageSource Source
+		{
+			get;
+			set;
+		}
+
+		public IMauiContext MauiContext
+		{
+			get;
+			set;
+		}
+
+		public bool IsAnimationPlaying { get; set; }
+		public bool IsLoading { get; private set; }
+
+		public void UpdateIsLoading(bool isLoading)
+		{
+			IsLoading = isLoading;
+		}
+
+		public static Task<IImageSourceServiceResult<NativeView>> GetImageAsync(IImageSource imageSource, IMauiContext mauiContext)
+		{
+			if (imageSource == null)
+				return Task.FromResult<IImageSourceServiceResult<NativeView>>(new ImageSourceServiceResult(null));
+
+			var services = mauiContext.Services;
```

Comment:
can we cache these calls?

Reply:
These are all static methods. We don't have anywhere in this context to cache anything.